libs/ledger-live-common/src/families/cosmos/js-synchronisation.ts

```diff
@@ -54,130 +54,110 @@ const txToOps = (info: AccountShapeInfo, accountId: string, txs: CosmosTx[]): Op
 
     const correspondingMessages = messages.filter(m => m.type === mainMessage.type);
 
-    if (correspondingMessages.length === 0) {
-      continue;
-    }
-
-    // TODO: This mechanism should be removed
-    const attributes: { [id: string]: any } = {};
-    mainMessage.attributes.forEach(item => (attributes[item.key] = item.value));
-
-    // https://docs.cosmos.network/v0.42/modules/staking/07_events.html
     switch (mainMessage.type) {
       case "transfer":
-        if (attributes.sender && attributes.recipient && attributes.amount) {
-          op.senders.push(attributes.sender);
-          op.recipients.push(attributes.recipient);
-
-          if (attributes.amount.indexOf(currency.units[1].code) != -1) {
-            op.value = op.value.plus(attributes.amount.replace(currency.units[1].code, ""));
-          }
-
-          if (!op.type && attributes.sender === address) {
-            op.type = "OUT";
-            op.value = op.value.plus(fees);
-          } else if (!op.type && attributes.recipient === address) {
-            op.type = "IN";
+        // TODO: handle IBC transfers here
+        for (const message of correspondingMessages) {
+          const amount = message.attributes.find(attr => attr.key === "amount")?.value;
+          const sender = message.attributes.find(attr => attr.key === "sender")?.value;
+          const recipient = message.attributes.find(attr => attr.key === "recipient")?.value;
```

Comment:
Are we sure there is always one recipient ? If we try to send to two recipient, will there be two message or two attributes ?

Reply:
In one "CosmosMessage" (event), in a send event, we have only one "sender" sender and one "recipient". 
If you want to send funds to multiple recipients, it is possible that have multiple events in one transactions.
we can refer to 
the cosmos sdk source code: https://github.com/cosmos/cosmos-sdk/blob/1b0fcdc9f059419328ed2b6b2393320fb042270f/x/bank/keeper/send.go#L141
to have a look at the source code of multi send transaction and make sure that one event can have only one sender and one recipient.

Reply:
Thanks, perfect